### DISCUSSION TITLE
rednet.recieve bug

### DISCUSSION THREAD

**Discussion by HappyYoyo09** [cc-tweaked/CC-Tweaked#1529] [Q&A] [answered]
### Minecraft Version

1.20.x

### Version

1.106.1

### Details

when using the code from the official wiki, it gives an error. the code in question is the rednet.recieve code with no timeout. it just says that there is an unexpected = sign, and suggests using ==, which does nothing.
here's a link to the log
https://mclo.gs/XQbYde0

**Comment by SquidDev** [comment]
Can you post the code you're trying to run? I've just tested [all three examples from the docs](https://tweaked.cc/module/rednet.html#v:receive), and they appear to work.

**Comment by HappyYoyo09** [comment]
sure, here it is

```lua
rednet.open(
--the rednet api returns the id of the computer
-- sending the message, then the message.
--this saves the id variable as "id" and the message
--as "message"
id,message=rednet.recieve
print(id)
print(message)
```

the sending code seems to have no issues, though here it is anyway
```lua
rednet.open("right")
rednet.send(12,"hello from the other side!" ) 
```
i have checked, the receiving computer is id 12, also i,m using a wired modem if that changes anything.

**Reply by SquidDev** [reply, marked as answer]
Oh gosh, I'm not sure where you got that code from, but it's horribly garbled. It's missing a couple of pretty key bits. This is what the code should look like:

```lua
rednet.open() -- <- Note the extra ) here. () indicates you're calling the function
--the rednet api returns the id of the computer
-- sending the message, then the message.
--this saves the id variable as "id" and the message
--as "message"
id,message=rednet.recieve() <-- Again, we need a () here to actually call the function.
print(id)
print(message)
```

**Reply by HappyYoyo09** [reply]
Omg thank you! As you can probably tell, I downloaded this mod yesterday, I know a bit of python, but when it comes to lua, I just kinda mess around and see what works 😅 I'll be sure to implement this next time I play.

**Reply by fatboychummy** [reply]
Do note you will need to pass a side to `rednet.open()` as well, so if the modem is on top you'd do `rednet.open("top")`. If you don't do this, the computer will pause as if it is listening for a message but never actually receive one because the modem isn't listening. The modem will have its edges turn red when it is actively listening for messages.
